subsys/bluetooth/host/gatt.c

```diff
@@ -2674,6 +2671,17 @@ uint8_t bt_gatt_check_perm(struct bt_conn *conn, const struct bt_gatt_attr *attr
 	}
 
 	mask &= attr->perm;
+
+	if (mask & BT_GATT_PERM_LESC_MASK) {
+#if defined(CONFIG_BT_SMP)
+		if ((conn->le.keys->flags & BT_KEYS_SC) == 0) {
+			return BT_ATT_ERR_INSUFFICIENT_ENCRYPTION;
+		}
+#else
+		return BT_ATT_ERR_INSUFFICIENT_ENCRYPTION;
```

Comment:
The error code is invalid. Core mandates to return Insufficient Authentication error in such cases.

Core Version 5.3 | Vol 3, Part F, 3.4.4.3 ATT_READ_REQ
"If the client has insufficient security to read the requested attribute then an
ATT_ERROR_RSP PDU shall be sent with the Error Code parameter set to
Insufficient Authentication (0x05)."

Insufficient Encryption error is returned in case the client has not enabled encryption which is not the case here.

Reply:
I am not sure whether we can rely on the core spec for the right error code here, as the LESC check is not covered by the core spec. 

The definition of the Insufficient Authentication is:
![image](https://user-images.githubusercontent.com/2520413/178694980-05b106d5-459b-4a3c-be3b-b22c125337e2.png)

This does not cover the LESC check - The authentication check is provide MITM protection.

Reply:
Any device that receives the insufficient authentication should attempt to upgrade the security level to 3 or 4 (as those provide authentication), but most LE Audio devices probably won't support that, as that would require IO capabilities.

Reply:
> Any device that receives the insufficient authentication should attempt to upgrade the security level to 3 or 4 (as those provide authentication), but most LE Audio devices probably won't support that, as that would require IO capabilities.

I could not find it in Core. Is that Core specification requirement?

On the other hand, the definition of the Insufficient Encryption is:
![image](https://user-images.githubusercontent.com/10126877/178703282-3b073c1e-b130-4445-8ad3-950a85cc1116.png)

Which does not cover this check as well - moreover
![image](https://user-images.githubusercontent.com/10126877/178704203-13ba6d32-0cc8-4f2f-b973-7c12aac9371f.png)

so this would be odd to return this error if the encryption is already enabled.

Reply:
@MariuszSkamra I don't necessarily see the conflict in your last text:
`If the client has not enabled encryption, and encryption is required to read the
requested attribute, then an ATT_ERROR_RSP PDU shall be sent with the
Error Code parameter set to Insufficient Encryption (0x0F).`

We can still comply with this, while also sending `Insufficient Encryption` as per implementation, right?